### PR TITLE
[FIX] small ICA bug fixes for #5967

### PR DESCRIPTION
**Pull request by jeythekey**
This PR already addresses the first two points in #5967.

Additionally, I have incorporate a missed suggestion from #5620 + a summary sentence and a reference of the valuable discussion there to the ICA docstring. (Because I unfortunately could not do this last time.)

### CODE REVIEW

**Comment by larsoner** [comment]
Can you add some test that would have shown this problem?

PEP8 problems

https://travis-ci.org/mne-tools/mne-python/jobs/496009578#L3207

**Comment by jeythekey** [comment]
Where should I add this test? Can I use some template for that?

**Comment by larsoner** [comment]
Look into https://github.com/mne-tools/mne-python/blob/master/mne/channels/tests/test_interpolation.py#L107

**Comment by jeythekey** [comment]
Thanks!!

**Comment by jeythekey** [comment]
Ah, that was for the interpolation test, but I understand. Should I modify those existing test files? Or should I create a special/temporary file for only the change of this PR? (Sorry, first time ...)

**Comment by larsoner** [comment]
I would add a new test function `test_interpolate_meg_ctf` right below those lines in the same file, and do the assertion there.

**Comment by larsoner** [comment]
Oh sorry I was commenting for the wrong PR :)

Perhaps it's not so easy to add a test for the filtering problem because you'd have to have a component in the pass-band or stop-band, then see that it was kept (or removed) with the correct `sfreq`. But can you easily show the indexing problem? Like you know the correct component should have been index 1 for example but it was showing up as 5 because of ordering issues?

**Comment by jeythekey** [comment]
I'll try my best and also write a test for the ica_apply problem with np.array ica.exclude next week.

File: mne/preprocessing/ica.py [hunk 1]
```diff
@@ -244,11 +244,15 @@ class ICA(ContainsMixin):
               ii) additional noise reduction, it is a matter of current debate
               whether pre-ICA dimensionality reduction could decrease the
               reliability and stability of the ICA, at least for EEG data and
-              especially during preprocessing [5]_.
+              especially during preprocessing [5]_. (But see also [6]_ for a
+              possibly confounding effect of the different whitening/sphering
+              methods used in this paper (ZCA vs. PCA).)
+              sphering is more commonly called ZCA
```
Comment (jona-sassenhagen) [inline mne/preprocessing/ica.py]: Start of a new sentence, right?

Comment (jeythekey) [reply mne/preprocessing/ica.py]: Ah, I forgot to clean that last line, it's an artefact from the composing process ;) Thanks. Or do you think it's worth to mention the nomenclatura confusions surrounding whitening/sphering/PCA/ZCA? I wouldn't do that here, is probably too long already anyway, right?

Comment (agramfort) [reply mne/preprocessing/ica.py]: yes. Just remove these lonely words and +1 for MRG

**Comment by jeythekey** [comment]
I just pushed the 2 fixes (PEP 8 and the lonely line) + the fix for ica.exclude as discussed in #5967. 
But @larsoner wanted me to write some tests for this, which I will do next week, so merging has to wait at least until then.

**Comment by jeythekey** [comment]
Ahhhh ... Again the rebase or whatever messed up everything. I seem to do something wrong. I'll again start with a new branch. Sorry ...

**Comment by jeythekey** [comment]
Continues in #6246 ...

**Pull request closed without merging**